### ISSUE TITLE
Adjust branching for tagging

### ISSUE THREAD

**Issue by cdnicoll** [ocupop/ocubits#60]
We're hitting an error with tagging action:

```
remote: error: GH006: Protected branch update failed for refs/heads/main.        
remote: error: Changes must be made through a pull request.        
To https://github.com/ocupop/ocubits
 * [new tag]         v0.2.9 -> v0.2.9
 ! [remote rejected] main -> main (protected branch hook declined)
error: failed to push some refs to 'https://github.com/ocupop/ocubits'
Error: Process completed with exit code 1.
```

https://github.com/ocupop/ocubits/actions/runs/7093165106/job/19306032808#step:6:11

**Comment by cdnicoll** [comment]
While we're at it, we might want to adjust this workflow and have it auto tag, and publish if a merge is completed into main.

**Comment by cdnicoll** [comment]
After speaking with Paul and Tom, I'll create a action strategy that once a PR is merged into main it will:
- [ ] create a minor tag version bump
- [ ] Publish to npm

These should be automated upon a successful merge.

If we want to have a patch or major versioning this should be done by a manual process
